### ISSUE TITLE
Inconsistent manifest.json makes it impossible to create a PWA

### ISSUE THREAD

**Issue by ElRocha** [aeharding/voyager#808]
**Describe the bug**
When trying to create a PWA in the browser from the site, an error occurs when trying to grab info from the `manifest.json` because the endpoint returns an empty app instead of the json file.

**To Reproduce**

1. Go to `https://vger.app/manifest.json` and an empty version of the app appears.
2. Full reload the page (ctrl + shift + r) and the expected json appears.
3. Regular reload (ctrl + r) and the empty app appears again.

**Expected behavior**
Always show the json file on that endpoint.

**Screenshots**
With ctrl + r:
![image](https://github.com/aeharding/voyager/assets/23618154/fa39193a-e7a3-4d7b-bd46-8daa14764f34)

With ctrl + shift + r:
![image](https://github.com/aeharding/voyager/assets/23618154/39f5ce63-64aa-4670-8258-e6fef9fed0c0)

**Desktop (please complete the following information):**

- OS: Windows 11
- Browser: Brave and Firefox
- Version: 1.59.120 (Brave), 118.0.2 (Firefox)

**Comment by aeharding** [comment]
Hi there! Can you show me the error? IME when browsers get the manifest.json file, they do not load the service worker so this shouldn't be a problem.

**Comment by ElRocha** [comment]
It seems that while the behaviour occurs in Brave as well, the issue when installing the web app only happens in Firefox with the extension.

Here's the error using the `Progressive Web Apps for Firefox` extension
![image](https://github.com/aeharding/voyager/assets/23618154/97d5a484-fb55-4a4a-9107-050c74d37729)

I did try to create another PWA (Whatsapp) and that worked fine in Firefox.

**Comment by aeharding** [comment]
I see, thanks for the info!

**Comment by arunsathiya** [comment]
>the issue when installing the web app only happens in Firefox with the extension.

@ElRocha Am I understanding correctly that the empty response occurs only when that extension is installed?

I have been testing a few times with Firefox and I see a proper response at https://vger.app/manifest.json every time, without requiring a hard refresh. Also fine on curl:

```shell
curl -LIX GET 'https://vger.app/manifest.json' -H "Cache-Control: no-cache" -A "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/118.0"
HTTP/2 200
server: nginx/1.25.1
date: Fri, 27 Oct 2023 21:41:18 GMT
content-type: application/json; charset=UTF-8
content-length: 1358
x-powered-by: Express
accept-ranges: bytes
cache-control: public, max-age=0
last-modified: Wed, 25 Oct 2023 05:25:13 GMT
etag: W/"54e-18b654b42a8"
vary: Accept-Encoding
x-frame-options: SAMEORIGIN
x-xss-protection: 1; mode=block
x-content-type-options: nosniff
referrer-policy: no-referrer-when-downgrade
content-security-policy: default-src * data: blob: 'unsafe-eval' 'unsafe-inline'
strict-transport-security: max-age=31536000; includeSubDomains; preload
```

**Comment by aeharding** [comment]
@arunsathiya right, the problem is that the service worker gets installed and the manifest.json request is being served from the service worker. In all cases up till now, manifest.json isn't served through the service worker. Just for these extensions, it seems.

**Comment by sharunkumar** [comment]
I was able to reproduce the blank page for manifest link on edge, using the following steps:

- open private window
- visit vger.app
- wait for the page to redirect to https://vger.app/posts/lemmy.world/all
- click the url, delete posts/lemmy.world/all, replace with manifest.json so now the url is https://vger.app/manifest.json
- hit enter

results in blank page every time

however, I made some local changes to `server.mjs` to always route /manifest.json` to serve the file, and I am no longer facing the issue (routed localhost dev server via cloudflare tunnels to check).

```diff
diff --git a/server.mjs b/server.mjs
index ad3ec47..efca0fc 100644
--- a/server.mjs
+++ b/server.mjs
@@ -2,6 +2,8 @@ import express from "express";
 import compression from "compression";
 import ViteExpress from "vite-express";
 import { createProxyMiddleware } from "http-proxy-middleware";
+import { fileURLToPath } from "url";
+import path from "path";
 
 const CUSTOM_LEMMY_SERVERS = process.env.CUSTOM_LEMMY_SERVERS
   ? process.env.CUSTOM_LEMMY_SERVERS.split(",").map((s) => s.trim())
@@ -34,9 +36,16 @@ INITIAL_VALID_LEMMY_SERVERS.forEach(
 const app = express();
 
 const PROXY_ENDPOINT = "/api/:actor";
+const PAW_MANIFEST = "/manifest.json";
 
 app.use(compression());
 
+app.get(PAW_MANIFEST, (req, res, next) => {
+  const __filename = fileURLToPath(import.meta.url);
+  const __dirname = path.dirname(__filename);
+  res.sendFile(path.join(__dirname, "public", "manifest.json"));
+});
+
 app.use(PROXY_ENDPOINT, async (req, res, next) => {
   const actor = req.params.actor;
```

@aeharding should I raise a PR with the changes?

**Comment by aeharding** [comment]
@sharunkumar You're not experiencing this on dev server because dev doesn't have the service worker installed.

I think I fixed this in #840 though!
